src/simulator/simulator.cpp

```diff
@@ -191,7 +263,47 @@ void Simulator::InitalLocationCallback(const PoseWithCovarianceStamped& msg) {
          cur_loc_.translation.x(),
          cur_loc_.translation.y(),
          RadToDeg(cur_loc_.angle));
-  motion_model_->SetPose({angle, loc});
+  motion_models_[car_num]->SetPose({angle, loc}); // TODO, add loop
+}
+
+void Simulator::InitalLocationCallbackCar0(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 0);
+}
+
+void Simulator::InitalLocationCallbackCar1(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 1);
+}
+
+void Simulator::InitalLocationCallbackCar2(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 2);
+}
+
+void Simulator::InitalLocationCallbackCar3(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 3);
+}
+
+void Simulator::InitalLocationCallbackCar4(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 4);
+}
+
+void Simulator::InitalLocationCallbackCar5(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 5);
+}
+
+void Simulator::InitalLocationCallbackCar6(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 6);
+}
+
+void Simulator::InitalLocationCallbackCar7(
+    const PoseWithCovarianceStamped &msg) {
+  InitalLocationCallbackMulti(msg, 7);
```

Comment:
Is there a way to write a single wrapper out of InitialLocationCallbackMulti?

Reply:
I have tried for roughly ~2-3 hours trying to find a workaround by creating higher-order functions using Boost/Std. Unfortunately, my code will not compile for some reason. I will take a look again but for now, I do not have a way of wrapping around the hardcoded functions.

Reply:
See my comment above. We can discuss.

Reply:
Sorry for the delay. I have looked into modifying ros messages to account for the instance of multiple cars. However, the message used (PoseWithCovarianceStamped) is one of the default messages that ros uses, which means modifying that would require everyone to manually perform the same modification.

As an alternative solution, I can define a custom message that accounts for multiple robots. Please let me know if you guys think that I should do that.